src/kicker/main.c

```diff
@@ -121,13 +126,15 @@ void main() {
         // get a voltage reading by weighing in a new reading, same concept as
         // TCP RTT estimates (exponentially weighted sum)
 
-        if (time % 400 == 0) {
+        // 10 us * 4000 = 40 ms, we don't need to run the ADC much faster
+        if (time % 4000 == 0) {
```

Comment:
Not 100% sure how time works, but is this only giving a 10 us window for this to run every 40 ms? and we have a 10 us delay in this loop. Maybe store the last time it ran?

Reply:
The intention was just to sporadically call the ADC read.